src/main/kotlin/edu/carroll/polyominoes/service/account/RegisterServiceImpl.kt

```diff
@@ -0,0 +1,120 @@
+package edu.carroll.polyominoes.service.account
+
+import edu.carroll.polyominoes.jpa.model.Account
+import edu.carroll.polyominoes.jpa.repo.AccountRepository
+import org.slf4j.LoggerFactory
+import org.springframework.security.crypto.bcrypt.BCrypt
+import org.springframework.stereotype.Service
+
+@Service
+class RegisterServiceImpl(private val accountRepo: AccountRepository) : RegisterService {
+
+    companion object {
+        private val log = LoggerFactory.getLogger(RegisterServiceImpl::class.java)
+    }
+
+    /**
+     * Given a username, determine if the user's given username already exist in the database.
+     *
+     * @param username - An username representing a user's account
+     * @return true if the user's given username is in the database, false otherwise
+     */
+    override fun validateUsernameExist(username: String): Boolean {
```

Comment:
I would consider renaming this method to something like 'usernameAvailable', so that if you get invalid information, you can return false, as well as return false if the username is taken, and the code is consistent and not misleading.  (See next comment)